File: CDC/NodeJS/sample_request.js
const axios = require("axios");
const FormData = require("form-data");
const qs = require("qs");
const fs = require("fs");
const https = require("https");

async function authorize(
  identityURL,
  clientID,
  clientSecret,
  username,
  password,
  grantType,
  apiKey
) {
  let data = qs.stringify({
    client_id: clientID,
    username: username,
    password: password,
    grant_type: grantType,
    client_secret: clientSecret,
  });

  try {
    console.log("Sending Request to Authorize");
    response = await axios.post(identityURL, data, {
      headers: {
        "x-api-key": apiKey,
        "Content-Type": "application/x-www-form-urlencoded",
      },
      httpsAgent: new https.Agent({ rejectUnauthorized: false }),
    });
    if (response.status == 200) {
      console.log("Authorized Successfully");
      return response.data.access_token;
    } else {
      console.log("Authorization Failed");
      return null;
    }
  } catch (error) {
    console.log(error);
    return null;
  }
}

async function classify(cdcURL, accessToken, apiKey, documentPath) {
  let data = new FormData();
  data.append("document", fs.createReadStream(documentPath));
  data.append("metadata", '{ "IHXPayerID": "<payer-id>"}');
  data.append(
    "process_info",
    '{"KYC_Classification": true,"Page_Error": "IGNORE"}'
  );

  try {
    console.log("Sending Request to Classify");
    response = await axios.post(cdcURL, data, {
      headers: {
        "Content-Type": "multipart/form-data",
        Accept: "application/json",
        "access-token": accessToken,
        "x-api-key": apiKey,
        ...data.getHeaders(),
      },
      httpsAgent: new https.Agent({ rejectUnauthorized: false }),
    });
    if (response.status == 200) {
      console.log("Classification Successful");
      return response.data;
    } else {
      console.log("Classification Failed");
      return null;
    }
  } catch (error) {
    console.log(error);
    return null;
  }
}

async function main(documentPath) {
  const clientID = "<client-id>";
  const clientSecret = "<client-secret>";
  const username = "<username>";
  const password = "<password>";
  const grantType = "<grant-type>";

  const identityURL = "https://<identity-url>";

  const cdcURL = "https://<cdc-url>";

  const apiKey = "<api-key>";

  try {
    accessToken = await authorize(
      identityURL,
      clientID,
      clientSecret,
      username,
      password,
      grantType,
      apiKey
    );
    if (accessToken != null) {
      response = await classify(cdcURL, accessToken, apiKey, documentPath);
      if (response != null) {
        return JSON.stringify(response);
      } else {
        return null;
      }
    }
  } catch (error) {
    console.log(error);
    return null;
  }
}

let response = new Promise((resolve, reject) => {
  try {
    resolve(main("<path-to-document>"));
  } catch (error) {
    reject(error);
  }
});

response.then((value) => {
  console.log(value);
});

File: CDC/Python/sample_request.py
import requests
from urllib.parse import quote_plus


# API for token generation
def authorize(
    identity_url, client_id, client_secret, username, password, grant_type, api_key
):
    """
    This function generates the access token for the user
    :param identity_url: Identity URL
    :param client_id: Client ID
    :param client_secret: Client Secret
    :param username: Username
    :param password: Password
    :param grant_type: Grant Type
    :param api_key: API Key
    :return: Access Token
    """
    print("Calling Authorizer")
    try:
        # need to escape special characters in password (if any)
        password = quote_plus(password)
        payload = f"client_id={client_id}&client_secret={client_secret}&username={username}&password={password}&grant_type={grant_type}"
        headers = {
            "x-api-key": api_key,
            "Content-Type": "application/x-www-form-urlencoded",
        }

        response = requests.request("POST", identity_url, headers=headers, data=payload,verify=False)
        print("Authorizer Response Code")
        print(response.status_code)
        if response.status_code == 200:
            return response.json()["access_token"]
        else:
            return None
    except Exception as e:
        print(e)
        return None


# DC Call API
def call_dc_api(url, access_token, api_key, payload, file_path):
    """
    This function calls the DC API
    :param url: DC URL
    :param access_token: Access Token
    :param api_key: API Key
    :param payload: Payload
    :param file_path: File Path
    :return: Response - JSON : Classification Result
    """
    print("Calling DC")
    try:
        file_name = file_path.split("/")[-1]
        files = [
            (
                "document",
                (
                    file_name,
                    open(
                        file_path,
                        "rb",
                    ),
                    "application/pdf",
                ),
            )
        ]
        headers = {
            "Accept": "application/json",
            "x-api-key": api_key,
            "access-token": access_token,
        }

        response = requests.request(
            "POST", url, headers=headers, data=payload, files=files,verify=False
        )
        if response.status_code == 200:
            print("Calling DC Successful")
            return response.json()
        else:
            print("Calling DC Failed")
            return None
    except Exception as e:
        print(e)
        return None


def execute_dc(file_path):
    """
    This function executes the DC API
    :param file_path: File Path
    :return: Response - JSON : Classification Result
    """
    client_id = "<client-id>"
    client_secret = "<client-secret>"
    username = "<username>"
    password = "<password>"
    grant_type = "<grant-type>"
    identity_url = "https://<identity-url>"
    dc_url = "https://<dc-url>"
    api_key = "api-key"
    payload = {
        "metadata": '{ "IHXPayerID": "<payer-id>"}',
        "process_info": '{"KYC_Classification": true,"Page_Error": "IGNORE"}',
    }
    try:
        access_token = authorize(
            identity_url,
            client_id,
            client_secret,
            username,
            password,
            grant_type,
            api_key,
        )
        if access_token is not None:
            return call_dc_api(dc_url, access_token, api_key, payload, file_path)
        else:
            return None
    except Exception as e:
        print(e)
        return None


dc_response = execute_dc("path-to-file")
print(dc_response)
